### PR TITLE
feat: add --lock option to remove command

### PR DESCRIPTION
**Pull request by ralbertazzi**
# Pull Request Check List

Resolves: #3971

Needs also https://github.com/python-poetry/poetry/pull/7915 to avoid the logging bug

### CODE REVIEW

**Comment by radoering** [comment]
We should add a test that checks:

- `remove` without `--lock` removes the dependency
- `remove --lock` does not remove the dependency

File: src/poetry/console/commands/remove.py [hunk 1]
```diff
@@ -108,15 +109,13 @@ def handle(self) -> int:
             )
 
         # Refresh the locker
-        self.poetry.set_locker(
-            self.poetry.locker.__class__(self.poetry.locker.lock, poetry_content)
```
Comment (ralbertazzi) [inline src/poetry/console/commands/remove.py:112]: There you go with the tests :) In order to make them work I had to work around this line that was effectively resetting the mocked attributes of the `TestLocker`

Comment (ralbertazzi) [reply src/poetry/console/commands/remove.py:112]: If you like it I will apply it to the `AddCommand` too

Comment (radoering) [reply src/poetry/console/commands/remove.py:112]: Looks better to me than before.

**Comment by radoering** [comment]
Now, we only have to mention the option in the [docs](https://python-poetry.org/docs/cli/#options-5).

**Review by radoering — APPROVED** [review]

**Pull request merged**